### DISCUSSION TITLE
Question: debugging memory usage

### DISCUSSION THREAD

**Discussion by Kakadu** [ocaml/ocaml-lsp#477] [Q&A] [answered]
My ocaml-lsp 1.4.1 still hits 60% of my 8Gb RAM from time to time. I'm wondering which are the sane approaches to debug memory usage of lsp. For example, could it be possible to report current memory usage of data structures after receiving SIGUSR1 or something....

**Comment by rgrinberg** [comment, marked as answer]
Perhaps you could use the new profiler to see what is being retained? https://blog.janestreet.com/finding-memory-leaks-with-memtrace/
